### PR TITLE
Fix private key authentication with dbshell

### LINKED ISSUE #67: Option private_key does not work when launching SnowSQL
The Python Connector accepts an option `private_key` to perform auth. For a server backend such as a Django app this is often the most preferable option.

The value for `private_key` [is accepted only as a `bytes` object](https://github.com/snowflakedb/snowflake-connector-python/blob/0894b7886793d92533c778d5becbd2f529418cc1/src/snowflake/connector/auth/keypair.py#L48-L60) that represents a [decrypted key's bytes](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-example#label-python-key-pair-authn-rotation).

The `OPTIONS` clause accepts passing a value for `private_key`, but it attempts to also reuse the same value as a file path when launching `snowsql`. Since the valid value can only be a raw bytes object, launching SnowSQL with its value for `--private-key-path` will fail (or conversely the regular connections in the server will fail if its a path):

```
Traceback (most recent call last):
  File "/django-tests/mysite/manage.py", line 22, in <module>
    main()
  File "/django-tests/mysite/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/django-tests/denv/lib/python3.10/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/django-tests/denv/lib/python3.10/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/django-tests/denv/lib/python3.10/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/django-tests/denv/lib/python3.10/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/django-tests/denv/lib/python3.10/site-packages/django/core/management/commands/dbshell.py", line 30, in handle
    connection.client.runshell(options["parameters"])
  File "/django-tests/denv/lib/python3.10/site-packages/django_snowflake/client.py", line 68, in runshell
    super().runshell(parameters)
  File "/django-tests/denv/lib/python3.10/site-packages/django/db/backends/base/client.py", line 28, in runshell
    subprocess.run(args, env=env, check=True)
  File "/3.10/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/3.10/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/3.10/lib/python3.10/subprocess.py", line 1796, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
ValueError: embedded null byte
```

I propose introducing special options `private_key_file` and `private_key_passphrase` that are specific to this project. Use of this internally auto-decrypts the file to the bytes format required by the connector (injecting `private_key` automatically into connector calls), and uses the same file path for its SnowSQL launch args.

Perhaps it makes sense to implement these directly into the connector itself (rather than in this plugin).

Happy to send a PR implementing this. Thoughts?

### PR DESCRIPTION
**Pull request by timgraham**
Fixes #67 

(Pending the release of snowflake-connector-python 3.6.0 which supports the new options.)

### CODE REVIEW

**Comment by timgraham** [comment]
@sfc-gh-hachouraria snowflake-connector-python 3.6.0 was just released. Could you test this?

File: django_snowflake/client.py [hunk 1]
```diff
@@ -33,17 +34,17 @@ def settings_to_cmd_args_env(cls, settings_dict, parameters):
...
-            args += ['--client-session-keep-alive', client_session_keep_alive]
+            args += ['--client-session-keep-alive']
         if dbname:
             args += ['-d', dbname]
         if host:
             args += ['-h', host]
         if passcode:
             args += ['--mfa-passcode', passcode]
         if passcode_in_password:
-            args += ['--mfa-passcode-in-password', passcode_in_password]
```
Comment (sfc-gh-hachouraria) [inline django_snowflake/client.py:44]: Was this removal (`passcode_in_password`) accidental?

Comment (timgraham) [reply django_snowflake/client.py:44]: No, see the second commit of the PR (it's not related to the private key stuff). This and client-session-keep-alive are boolean options that don't take arguments.

**Comment by timgraham** [comment]
Thanks for raising the issue about an inappropriately required PASSWORD. I added a commit to address it.

**Comment by sfc-gh-hachouraria** [comment]
> Thanks for raising the issue about an inappropriately required PASSWORD. I added a commit to address it.

Thank you! I had no idea internal configs already existed for skipping that requirement — Django is brilliant.

Confirming I can completely skip `PASSWORD` in the config now with this change.

**Pull request merged**